### ISSUE TITLE
Try new Outlines logits processor wrapper for thinking models

### ISSUE THREAD

**Issue by lmmx** [lmmx/r1-cli#1]
See gist demoing the new functionality by Cameron Pfiffer: https://gist.github.com/cpfiffer/e98fc71d4fcb35ba827fc9e679112895
